tests/resource/aws_elasticache_cluster: TestAccAWSElasticacheCluster_ReplicationGroupID_SingleReplica Failures

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Description

We recently upgraded to Terraform Plugin SDK version 2, which uses actual Terraform binaries and command outputs for testing rather than relying on an (aging/divergent) internal Terraform core implementation in the SDK, mostly based on Terraform 0.11 with some shims for 0.12 handling. The shims in the previous version of the acceptance testing framework automatically accepted/consolidated resource addresses of `count = 1` to be `resource_type.resource_name` rather than the more correct `resource_type.resource_name.0` in Terraform 0.12 and later, which is now being used and expected. It appears this testing is not accounting for that difference:

```
=== CONT  TestAccAWSElasticacheCluster_ReplicationGroupID_SingleReplica
TestAccAWSElasticacheCluster_ReplicationGroupID_SingleReplica: resource_aws_elasticache_cluster_test.go:715: Check 2/6 error: Not found: aws_elasticache_cluster.test
--- FAIL: TestAccAWSElasticacheCluster_ReplicationGroupID_SingleReplica (1225.90s)
```

To not affect other testing, the fix for this should theoretically be as simple as adding `.0` to the end of the `clusterResourceName` variable:

https://github.com/terraform-providers/terraform-provider-aws/blob/d148d85330ff4cc95700043d8e6947fa71db7f2f/aws/resource_aws_elasticache_cluster_test.go#L712

### Affected Resource(s)

* aws_elasticache_cluster

### Definition of Done

* [ ] `TestAccAWSElasticacheCluster_ReplicationGroupID_SingleReplica` testing passes

### References

* [Contributing Guide on Running/Writing Acceptance Testing](https://github.com/terraform-providers/terraform-provider-aws/blob/master/docs/contributing/running-and-writing-acceptance-tests.md)

--- a/aws/resource_aws_elasticache_cluster_test.go
+++ b/aws/resource_aws_elasticache_cluster_test.go
@@ -709,7 +709,7 @@ func TestAccAWSElasticacheCluster_ReplicationGroupID_SingleReplica(t *testing.T)
 	var cluster elasticache.CacheCluster
 	var replicationGroup elasticache.ReplicationGroup
 	rName := acctest.RandomWithPrefix("tf-acc-test")
-	clusterResourceName := "aws_elasticache_cluster.test"
+	clusterResourceName := "aws_elasticache_cluster.test.0"
 	replicationGroupResourceName := "aws_elasticache_replication_group.test"
 
 	resource.ParallelTest(t, resource.TestCase{

